app/src/main/java/com/example/playlistmaker/activities/SearchActivity.kt

```diff
@@ -0,0 +1,144 @@
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ImageView
+import android.widget.LinearLayout
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.playlistmaker.R
+import com.example.playlistmaker.recyclerView.Track
+import com.example.playlistmaker.recyclerView.TrackAdapter
+import com.example.playlistmaker.retrofit.ITunesApi
+import com.example.playlistmaker.retrofit.TracksResponse
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class SearchActivity : AppCompatActivity() {
+
+    private val imdbBaseUrl = "https://itunes.apple.com"
+    private val retrofit = Retrofit.Builder()
+        .baseUrl(imdbBaseUrl)
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+    private val imdbService = retrofit.create(ITunesApi::class.java)
+    private val trackList = ArrayList<Track>()
+    private var restoredText = ""
+    private lateinit var recycler: RecyclerView
+    val adapter = TrackAdapter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContentView(R.layout.activity_search)
+        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+
+        findViewById<ImageView>(R.id.backIcon_search_screen).setOnClickListener { finish() }
+
+        val inputText = findViewById<EditText>(R.id.search_input_text)
+        recycler = findViewById(R.id.search_list)
+        val nothingImage = findViewById<LinearLayout>(R.id.nothingFound)
+        val connectionProblemError = findViewById<LinearLayout>(R.id.connectionProblem)
+        val clearButton = findViewById<ImageView>(R.id.search_clear_button)
+        val reloadButton = findViewById<Button>(R.id.reload_button)
+
+        if (savedInstanceState != null) inputText.setText(restoredText)
+
+        fun searchAction() {
+            imdbService.search(inputText.text.toString())
+                .enqueue(object : Callback<TracksResponse> {
+                    override fun onResponse(
+                        p0: Call<TracksResponse>,
+                        response: Response<TracksResponse>
+                    ) {
+                        if (response.code() == 200) {
```

Comment:
```suggestion
                        if (response.isSuccessfull) {
```
Лучше так проверять, не только 200 код говорит о том, что запрос успешно завершился